crates/egui-wgpu/src/renderer.rs

```diff
@@ -60,10 +66,26 @@ impl CallbackFn {
         Self::default()
     }
 
-    /// Set the prepare callback
+    /// Set the prepare callback.
+    ///
+    /// The passed-in `CommandEncoder` is egui's and can be used directly to register
+    /// wgpu commands for simple use cases.
+    ///
+    /// For more complicated use cases, one can also return a list of arbitrary
+    /// `CommandBuffer`s and have complete control over they get created and fed.
+    /// In particular, this gives an opportunity to parallelize command registration and
+    /// prevents a faulty callback from poisoning the main wgpu pipeline.
+    ///
+    /// The main egui command buffer, as well as all user-defined command buffers returned
+    /// by this function, are guaranteed to all be submitted at once in a single call.
```

Comment:
I really like your description here, but wonder if the doc block at the beginning should be shortened accordingly

Reply:
one detail though:
```
    /// The main egui command buffer, as well as all user-defined command buffers returned
    /// by this function, are guaranteed to all be submitted at once in a single call.
```

If someone uses the `Renderer` without eframe/egui_wgpu::winit.rs, it is their responsibility to do so. So it's less a guarantee. So maybe
```
    /// by this function, can then be submitted at once in a single draw call (as done by eframe).
```